### PR TITLE
debian/ubuntu: Configure apt chroot with APT_CONFIG

### PR DESCRIPTION
**Pull request by gdiscry**
I had a regression when updating from mkosi v13 to mkosi v14.

I'm building some images based on Debian Bookworm (currently testing) and use [snapshot.debian.org](https://snapshot.debian.org/) as mirror to avoid breaking the build as Bookworm is still in flux (for example, systemd-resolved was recently split into its own package which broke mkosi v13 if systemd-resolved is not in the Package list).

To stop apt from failing when the snapshot is older than 12 days, I put a file in `mkosi.skeleton/etc/apt/apt.conf.d/` with `Acquire::Check-Valid-Until "false";`.

When invoking apt switched from running inside the container to running on the host, apt stopped reading the configuration files present in the container. Because of this, it is not possible anymore to configure apt by dropping a file in `mkosi.skeleton/etc/apt/apt.conf.d/`.

apt first loads its configuration files and then parse the command-line options (so that they can override the options set in the files). While setting `-o Dir=<container-root>` on the command-line has some impact on the configuration of apt, it will not read the files in `<container-root>/etc/apt/apt.conf.d/`, as apt has already read its configuration files from the host with the configuration `Dir "/"`.

Rather than using the command-line, the APT_CONFIG environment variable should be used to point to a file with the chroot configuration. When set, that file is processed before anything else. By setting `Dir` this way, apt will read the configuration files from the container and not from the host.

As the host configuration is not read anymore, there is no more need to mask `/etc/apt` on the host by bind-mounting an empty directory over it.

### CODE REVIEW

File: mkosi/__init__.py [hunk 1]
```diff
@@ -2472,6 +2472,18 @@ def mount_apt_local_mirror(state: MkosiState) -> Iterator[None]:
             yield
 
 
+def setup_apt(state: MkosiState) -> None:
+    config_file = state.workspace / "apt.conf"
```
[the remark this answers is not in the document]
Comment (gdiscry) [reply mkosi/__init__.py:2476]: I was inspired by `setup_dnf` (which is slightly more complex 😉), but I also felt icky with the hidden dependency between `setup_{dnf,apt}` and `invoke_{dnf,apt}`.

**Review by bluca — APPROVED** [review]

**Review by DaanDeMeyer — APPROVED** [review]

**Comment by DaanDeMeyer** [comment]
@gdiscry Do you know if we can somehow configure apt's cache directories to be outside of the chroot directory? It would be amazing if we didn't have to mount those directories inside of the image when running apt

**Comment by DaanDeMeyer** [comment]
@gdiscry Thanks! I just realized we'd still need to figure out how to get the local repository (`file:<path>`) into the image without mounting though. I'm not sure if it's possible to specify a repository via a path outside of the chroot directory.

@bluca Could we spawn a HTTP server on localhost to serve the local mirror repository to avoid having to mount it? OBS should be fine with localhost network traffic right?

**Comment by bluca** [comment]
yeah that should be fine

**Pull request merged**